server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetStoredScriptAction.java

```diff
@@ -18,33 +18,61 @@
  */
 package org.elasticsearch.rest.action.admin.cluster;
 
+import org.apache.logging.log4j.LogManager;
 import org.elasticsearch.action.admin.cluster.storedscripts.GetStoredScriptRequest;
+import org.elasticsearch.action.admin.cluster.storedscripts.GetStoredScriptResponse;
 import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.logging.DeprecationLogger;
 import org.elasticsearch.rest.BaseRestHandler;
 import org.elasticsearch.rest.RestController;
 import org.elasticsearch.rest.RestRequest;
-import org.elasticsearch.rest.action.RestStatusToXContentListener;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.rest.action.RestToXContentListener;
+import org.elasticsearch.script.StoredScriptSource;
 
 import java.io.IOException;
+import java.util.Map;
 
 import static org.elasticsearch.rest.RestRequest.Method.GET;
+import static org.elasticsearch.rest.RestRequest.Method.HEAD;
+import static org.elasticsearch.rest.RestStatus.NOT_FOUND;
+import static org.elasticsearch.rest.RestStatus.OK;
 
 public class RestGetStoredScriptAction extends BaseRestHandler {
 
+    private static final DeprecationLogger deprecationLogger =
+        new DeprecationLogger(LogManager.getLogger(RestGetStoredScriptAction.class));
+
     public RestGetStoredScriptAction(RestController controller) {
-        controller.registerHandler(GET, "/_scripts/{id}", this);
+        controller.registerHandler(GET, "/_script", this);
+        controller.registerWithDeprecatedHandler(GET, "/_script/{id}", this,
+            GET, "/_scripts/{name}", deprecationLogger);
```

Comment:
I don't think this works to have the same location within a path have different names. The same position needs to use the same substitution.

Reply:
Thanks for the review. I should have stated that I sledgehammered in backwards compatibility, but did not finish due to the various issues. Also wanted a full build test.

I think my integration tests issues might be related to the comment above. By location, you mean the position in the path? How can the old path be deprecated then? I noticed that the typed action paths are not deprecated, which would explain integration tests not failing.

Reply:
Both the old and new paths can use the same variable name. Upon looking at this again, it isn't strictly necessary (although still recommended). The edge case I was referring to was if you have the same path name, but different variable names based on position eg `_script/{id}` and `_script/{name}/{context}`.

Reply:
I am only now seeing the issue. Merged two branches into the PR branch and missed it. Just what is Ryan talking about? Duh!